can/logconvert.py

```diff
@@ -0,0 +1,68 @@
+"""
+Convert a log file from one format to another.
+"""
+
+import sys
+import argparse
+import errno
+
+from can import LogReader, Logger, SizedRotatingLogger
+
+
+class ArgumentParser(argparse.ArgumentParser):
+    def error(self, message):
+        self.print_help(sys.stderr)
+        self.exit(errno.EINVAL, "%s: error: %s\n" % (self.prog, message))
+
+
+def main():
+    parser = ArgumentParser(
+        description="Convert a log file from one format to another.",
+    )
+
+    parser.add_argument(
+        "-s",
+        "--file_size",
+        dest="file_size",
+        type=int,
+        help="Maximum file size in bytes. Rotate log file when size threshold is reached.",
+        default=None,
+    )
+
+    parser.add_argument(
+        "input",
+        metavar="INFILE",
+        type=str,
+        help="Input filename. The type is dependent on the suffix, see can.LogReader.",
+    )
+
+    parser.add_argument(
+        "output",
+        metavar="OUTFILE",
+        type=str,
+        help="Output filename. The type is dependent on the suffix, see can.Logger.",
+    )
+
+    args = parser.parse_args()
+
+    reader = LogReader(args.input)
+
+    if args.file_size:
+        logger = SizedRotatingLogger(
+            base_filename=args.output, max_bytes=args.file_size
+        )
+    else:
+        logger = Logger(filename=args.output)
+
+    try:
+        for m in reader:  # pylint: disable=not-an-iterable
+            logger(m)
+    except KeyboardInterrupt:
+        sys.exit(1)
+    finally:
+        reader.stop()
+        logger.stop()
```

Comment:
I think this should do, it's a couple of lines shorter and maybe a little bit easier to understand:

```suggestion
    with LogReader(args.input) as reader:

        if args.file_size:
            logger = SizedRotatingLogger(
                base_filename=args.output, max_bytes=args.file_size
            )
        else:
            logger = Logger(filename=args.output)

        with logger:
            try:
                for m in reader:  # pylint: disable=not-an-iterable
                    logger(m)
            except KeyboardInterrupt:
                sys.exit(1)
```

Reply:
Just tested it, but `SizedRotatingLogger` doesn't implement `__enter__` so `python -m can.logconvert -s 1000 log.log log.asc` doesn't work.

Reply:
Whoops! That's a bug, I suppose. `SizedRotatingLogger` should inherit from `Logger`, which would then work since `BaseIOHandler` implements the context manager protocol.